constraints_test.go

```diff
@@ -166,11 +166,28 @@ func TestConstraintCheck(t *testing.T) {
 		{"<=1.1", "0.1.0", true},
 		{"<=1.1", "1.1.0", true},
 		{"<=1.1", "1.1.1", false},
+		{"<=1.1-alpha1", "1.1", false},
+		{"<=2.x", "3.0.0", false},
+		{"<=2.x", "2.9.9", true},
+		{"<2.x", "2.0.0", false},
```

Comment:
Maybe, or maybe I'm missing something. The semantics I'm operating on interpret `x` as "any nonnegative integer". So I take `<2.x` to mean "less than ANY possible valid value for `2.x`". `2.0.0` is a valid possible value for `2.x`, so it should fail.

No? I haven't actually looked at other semver libs to see how they deal with this, or if they even allow gt/lt with wildcard operators at all.